src/bun.js/bindings/KeyObject.cpp

```diff
@@ -0,0 +1,2385 @@
+        if (result.key) {
+            return result;
+        }
+    }
+    ERR_clear_error();
+    BIO_reset(bp.get());
+
+    // X.509 fallback.
+    if (PEM_bytes_read_bio(&result.der_data, &result.der_len, nullptr, "CERTIFICATE", bp.get(), nullptr, nullptr) == 1) {
+        const unsigned char* p = result.der_data;
+        X509Ptr x509(d2i_X509(nullptr, &p, result.der_len));
+        result.key = x509 ? X509_get_pubkey(x509.get()) : nullptr;
+        if (result.key) {
+            return result;
+        }
+        OPENSSL_clear_free(result.der_data, result.der_len);
+        ERR_clear_error();
+        result.der_data = nullptr;
+        result.der_len = 0;
+    } else {
+        OPENSSL_clear_free(result.der_data, result.der_len);
+        ERR_clear_error();
+        result.der_data = nullptr;
+        result.der_len = 0;
+    }
+    return result;
+}
+
+JSC::EncodedJSValue KeyObject__createPrivateKey(JSC::JSGlobalObject* globalObject, JSC::CallFrame* callFrame)
+{
+
+    auto count = callFrame->argumentCount();
+    auto& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    if (count < 1) {
+        JSC::throwTypeError(globalObject, scope, "createPrivateKey requires 1 arguments"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+
+    auto* options = jsDynamicCast<JSC::JSObject*>(callFrame->argument(0));
+    if (!options) {
+        JSC::throwTypeError(globalObject, scope, "expected options to be a object"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+
+    JSValue keyJSValue = options->getIfPropertyExists(globalObject, PropertyName(Identifier::fromString(vm, "key"_s)));
+    if (keyJSValue.isUndefinedOrNull() || keyJSValue.isEmpty()) {
+        JSC::throwTypeError(globalObject, scope, "key is required"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+    if (!keyJSValue.isCell()) {
+        JSC::throwTypeError(globalObject, scope, "key must be a Buffer, Array-like or object"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+
+    JSValue formatJSValue = options->getIfPropertyExists(globalObject, PropertyName(Identifier::fromString(vm, "format"_s)));
+    if (formatJSValue.isUndefinedOrNull() || formatJSValue.isEmpty()) {
+        JSC::throwTypeError(globalObject, scope, "format is required"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+
+    if (!formatJSValue.isString()) {
+        JSC::throwTypeError(globalObject, scope, "format must be a string"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+    auto format = formatJSValue.toWTFString(globalObject);
+    RETURN_IF_EXCEPTION(scope, encodedJSValue());
+
+    Zig::GlobalObject* zigGlobalObject = reinterpret_cast<Zig::GlobalObject*>(globalObject);
+    auto* structure = zigGlobalObject->JSCryptoKeyStructure();
+
+    void* data;
+    size_t byteLength;
+
+    auto type = keyJSValue.asCell()->type();
```

Comment:
We do multiple calls to `asCell()` here, we could just do one

Reply:
Fixed on https://github.com/oven-sh/bun/pull/5940/commits/59ebd2d62bf1da38db66be1cb3e00fcd2c56d302